File: Instala/ActiveX/ActiveX/exemplos/sitef/fiscal/c#/ValoresFrm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace exemplo
{
    public partial class ValoresFrm : Form
    {
        private int internalTamanhoMinimo = 0;
        
        public ValoresFrm()
        {
            InitializeComponent();
        }

        public int tamanhoMinimo
        {
            get
            {
                return internalTamanhoMinimo;
            }
            set
            {
                internalTamanhoMinimo = value;
            }
        }

        public int tamanhoMaximo
        {
            get
            {
                return txtValor.MaxLength;
            }
            set
            {
                txtValor.MaxLength = value;
            }
        }

        public bool ligarModoSenha
        {
            set
            {
                if (value)
                    txtValor.PasswordChar = '*';
            }
        }


        public void escreverMensagem(string msg)
        {
            lblMsg.Text = msg;
        }

        public string dadoLido
        {
            get
            {
                return txtValor.Text;
            }
        }

        private void btnFechar_Click(object sender, EventArgs e)
        {
            this.DialogResult = DialogResult.Cancel;
            this.Close();
        }

        private void btnOK_Click(object sender, EventArgs e)
        {
            if (txtValor.Text != "" && txtValor.Text.Length < internalTamanhoMinimo)
            {
                txtValor.Focus();
                MessageBox.Show("Por favor, preencha corretamente o campo,",
                    "Preenchimento", MessageBoxButtons.OK, MessageBoxIcon.Warning);
                return;
            }

            this.DialogResult = DialogResult.OK;
            this.Close();
        }

        private void ValoresFrm_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape)
            {
                this.DialogResult = DialogResult.Cancel;
                this.Close();
            }
        }

        private void btnTelaAnterior_Click(object sender, EventArgs e)
        {
            this.DialogResult = DialogResult.Retry;
            this.Close();
        }
    }
}

File: Instala/ActiveX/ActiveX/exemplos/sitef/nao fiscal/vb.net/NaoFiscal/NaoFiscal/ValoresFrm.vb
Public Class ValoresFrm
    Private InternalValorMinimo As Integer = 0

    Public Property ValorMinimo As Integer
        Get
            Return InternalValorMinimo
        End Get
        Set(ByVal value As Integer)
            InternalValorMinimo = value
        End Set
    End Property

    Public Property ValorMaximo As Integer
        Get
            Return txtValor.MaxLength
        End Get
        Set(ByVal value As Integer)
            txtValor.MaxLength = value
        End Set
    End Property

    Public WriteOnly Property LigarModoSenha As Boolean
        Set(ByVal value As Boolean)
            If (value) Then
                txtValor.PasswordChar = "*"
            Else
                txtValor.PasswordChar = ""
            End If
        End Set
    End Property

    Public Sub EscreverMensagem(ByVal Msg As String)
        lblMsg.Text = Msg
    End Sub

    Public ReadOnly Property DadoLido As String
        Get
            Return txtValor.Text
        End Get
    End Property

    Private Sub btnTelaAnterior_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnTelaAnterior.Click
        Me.DialogResult = Windows.Forms.DialogResult.Retry
        Me.Close()
    End Sub

    Private Sub btnOK_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOK.Click
        If (txtValor.Text <> "" And txtValor.Text.Length < InternalValorMinimo) Then
            txtValor.Focus()
            MessageBox.Show("Por favor, preencha corretamente o campo,",
                    "Preenchimento", MessageBoxButtons.OK, MessageBoxIcon.Warning)
        Else
            Me.DialogResult = Windows.Forms.DialogResult.OK
            Me.Close()
        End If
    End Sub

    Private Sub btnFechar_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnFechar.Click
        Me.DialogResult = Windows.Forms.DialogResult.Cancel
        Me.Close()
    End Sub

    Private Sub btnFechar_KeyUp(ByVal sender As System.Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles btnFechar.KeyUp
        If (e.KeyCode = Keys.Escape) Then
            Me.DialogResult = Windows.Forms.DialogResult.Cancel
            Me.Close()
        End If
    End Sub
End Class